src/main/java/com/amazonaws/encryptionsdk/AwsCrypto.java

```diff
@@ -270,14 +314,54 @@ public long estimateCiphertextSize(
     /**
      * Returns the equivalent to calling {@link #encryptData(CryptoMaterialsManager, byte[], Map)} with
      * an empty {@code encryptionContext}.
+     *
+     * @deprecated Replaced by {@link #encryptData(AwsCryptoConfig, byte[])}
      */
+    @Deprecated
     public CryptoResult<byte[], ?> encryptData(
             final CryptoMaterialsManager materialsManager,
             final byte[] plaintext
     ) {
         return encryptData(materialsManager, plaintext, EMPTY_MAP);
     }
 
+    /**
+     * Returns an encrypted form of {@code plaintext} that has been protected with either the {@link CryptoMaterialsManager}
+     * or the {@link Keyring} specified in the {@link AwsCryptoConfig}.
+     *
+     * @param config The AwsCryptoConfig containing either a {@link CryptoMaterialsManager} or a {@link Keyring} and
+     *               optionally an EncryptionContext
+     * @param plaintext The plaintext to encrypt
+     * @return An {@link AwsCryptoResult} containing the encrypted data
+     */
+    public AwsCryptoResult<byte[]> encryptData(final AwsCryptoConfig config,
```

Comment:
Since we're fundamentally changing the footprint of this API, I would love to take this opportunity to rename these to simply `encrypt` and `decrypt`. This is something I've been wanting to do for a long time but never had a justification. Doing this now has the benefit not only of simplifying the API but also making it explicitly clear what calls are impacted by that change not only by secondary identifiers like the return and parameter types, but by the method name itself.